sources/common/toddl.go

```diff
@@ -374,3 +374,31 @@ func CvtIndexHelper(conv *internal.Conv, tableId string, srcIndex schema.Index,
 	}
 	return spIndex
 }
+
+func spannerSchemaApplyExpressions(conv *internal.Conv, expressions internal.VerifyExpressionsOutput) {
+	for _, expression := range expressions.ExpressionVerificationOutputList {
+		switch expression.ExpressionDetail.Type {
+		case "DEFAULT":
+			{
+				tableId := expression.ExpressionDetail.Metadata["TableId"]
+				columnId := expression.ExpressionDetail.Metadata["ColId"]
+
+				if expression.Result {
+					col := conv.SpSchema[tableId].ColDefs[columnId]
```

Comment:
Yes, this is called when the initial migration is done, hence all columns and tables are migrated

Reply:
Can you point me to the caller of this method? I dont think we need to make the assumption that this will be always called during the initial migration.

We can check if the tableId and columnId is present in spSchema before trying to read the `col` object here. Then we can call this utility method in other flows as well (for example during expr validation on click of prepare migration)

Reply:
https://github.com/GoogleCloudPlatform/spanner-migration-tool/pull/960/files#diff-d26a50b05c1e5c3bf6a8442acd55441b32da1dec3ae486708c0ae2906da91a6fR79-R86